### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'replace')

### ISSUE THREAD

**Issue by ghost** [HashLips/hashlips_art_engine#1063]
Hi, i have this type error when i generate, any idea ?
Thanks
![bd](https://user-images.githubusercontent.com/103067553/162587211-73a29d9d-af6f-4537-8e5e-25f47a808ba2.PNG)

**Comment by bolshoytoster** [comment]
@ADRIhein2 could you try using [my fork](https://github.com/bolshoytoster/hashlips_art_engine) please.

**Comment by ghost** [comment]
I have all my changes in this one I'm afraid of losing everything

**Comment by bolshoytoster** [comment]
@ADRIhein2 you can copy everything over.

**Comment by ghost** [comment]
is yours exactly the same ?

**Comment by bolshoytoster** [comment]
@ADRIhein2 basically.

**Comment by ghost** [comment]
i found something, it only does this when I use another folders in my layers folder
![fez](https://user-images.githubusercontent.com/103067553/162589500-f6d9a0dc-4db8-4414-86ba-53ef9fe753f8.PNG)

**Comment by bolshoytoster** [comment]
@ADRIhein2 yes, it tries to load every file in the folder as an image,  even if it's not an image. My fork adds handling for this.

**Comment by ghost** [comment]
![yj](https://user-images.githubusercontent.com/103067553/162589936-f40795d1-53d2-44d8-b9bd-1bb333b0ed31.PNG)

**Comment by bolshoytoster** [comment]
@ADRIhein2 it looks like folders have incorrect permissions, try running visual studio as admin.

**Comment by ghost** [comment]
nop same

**Comment by bolshoytoster** [comment]
@ADRIhein2 you'll have to delete the `build/` folder yourself as well.

**Comment by ghost** [comment]
![tgy](https://user-images.githubusercontent.com/103067553/162590285-5a018505-fd6e-4d0a-b2f6-c5107edc269a.PNG)

**Comment by ghost** [comment]
always the same

**Comment by bolshoytoster** [comment]
@ADRIhein2 that's not my fork anyway.

Could you
1. Make a copy of your `src/config.js` and `layers/`.
2. Delete the `hashlips_art_engine-main/` folder.
3. [Download my fork](https://github.com/bolshoytoster/hashlips_art_engine/archive/refs/heads/main.zip).
4. Unzip it.
5. Run visual studio as admin.
6. Open the folder you extracted.
7. Copy over your saved `src/config.js` and `layers/`.

**Comment by ghost** [comment]
![Uploading rh.PNG…]()

**Comment by ghost** [comment]
![rh](https://user-images.githubusercontent.com/103067553/162590708-8a149a57-010a-4dac-a0c4-8502e199903a.PNG)

**Comment by bolshoytoster** [comment]
@ADRIhein2 you're in the wrong folder then, run `Get-ChildItem -Directory` to get the name, and run `cd ` + name to change into it.

**Comment by ghost** [comment]
![ujio](https://user-images.githubusercontent.com/103067553/162590855-1da3921f-983d-463c-842a-d3a4f9f82c79.PNG)
  isnt the same ?

**Comment by bolshoytoster** [comment]
@ADRIhein2 run `cd hashlips_art_engine-main`.

**Comment by ghost** [comment]
![jtn](https://user-images.githubusercontent.com/103067553/162590964-b0f87bc7-193b-4f87-8951-cb142fb8cf3b.PNG)

**Comment by bolshoytoster** [comment]
@ADRIhein2 it might be because your `layersOrder`s have `Eyes` instead of `Eyes/Four`.

**Comment by ghost** [comment]
it's on purpose
i want my first tree to have all Eyes ("Three" and "Four" folder) then my second to have only "Four" folder
![rge](https://user-images.githubusercontent.com/103067553/162591355-60dd9263-4715-4839-9125-7c95e56706f9.PNG)

**Comment by bolshoytoster** [comment]
@ADRIhein2 I'll add support for this now.

**Comment by ghost** [comment]
thank u !

**Comment by bolshoytoster** [comment]
@ADRIhein2 you can either download my repo again or add:
```js
       if (fs.statSync(`${path}/${name}/${i}`).isDirectory()) {
         elements.concat(getElements(path, `${name}/${i}`));
         return;
       }
```
between these two lines:
```js
     .forEach(i => {
       if (i.includes(DNA_DELIMITER)) {
```
In `src/main.js` ~line 107.

**Comment by ghost** [comment]
I don't know if I did right
![hte](https://user-images.githubusercontent.com/103067553/162592055-bdd69f97-313a-422f-bca2-11554ac7946e.PNG)

**Comment by bolshoytoster** [comment]
@ADRIhein2 that's right, you'll still have to delete `build/` yourself.

**Comment by ghost** [comment]
i did it then :
![rfhsd](https://user-images.githubusercontent.com/103067553/162592136-f3da29c9-0267-4a21-8d99-a7606e084ffc.PNG)

**Comment by bolshoytoster** [comment]
@ADRIhein2 could you try adding `console.log(layerConfigs, element)` to `src/main.js` line 293.

**Comment by ghost** [comment]
![zfsg](https://user-images.githubusercontent.com/103067553/162592308-3a08539a-b04a-4bee-afde-071e2425df34.PNG)

**Comment by bolshoytoster** [comment]
@ADRIhein2 delete `build/` again.

**Comment by ghost** [comment]
i did it but it reappears every time I generate

**Comment by bolshoytoster** [comment]
@ADRIhein2 try running `Remove-Item -Recurse build`.

**Comment by ghost** [comment]
Cannot find path 'C:\Users\PC Adrien\Desktop\hashlips_art_engine-main\build' because it does not exist.
![Uploading dgn.PNG…]()

**Comment by ghost** [comment]
![dgn](https://user-images.githubusercontent.com/103067553/162592571-c32e9afa-6f53-4bbd-9bb3-538cc563ebef.PNG)

**Comment by bolshoytoster** [comment]
@ADRIhein2 run `cd hashlips_art_engine-main` again.

**Comment by ghost** [comment]
![drhd](https://user-images.githubusercontent.com/103067553/162592640-f3276c17-b383-462a-a841-ea52517bf973.PNG)

**Comment by ghost** [comment]
a lot of theses lines now

**Comment by ghost** [comment]
![ryhe](https://user-images.githubusercontent.com/103067553/162592659-7fa4a4fa-e9d8-4144-b0f3-e4d13c49adc9.PNG)

**Comment by bolshoytoster** [comment]
@ADRIhein2 you can remove the `console.log` line now, could you change the line before:
```js
    layer = layerConfigs[element.split('/').shift()];
```
to
```js
    layer = layerConfigs[element.split('/').slice(0, -1).join('/')];
```

**Comment by ghost** [comment]
where can i find this ?

**Comment by bolshoytoster** [comment]
@ADRIhein2 line 294 in `src/main.js`.

**Comment by ghost** [comment]
![ertyh](https://user-images.githubusercontent.com/103067553/162593017-29a04d71-44ab-4e43-b7c3-a1a305892564.PNG)

**Comment by bolshoytoster** [comment]
@ADRIhein2 try running `Remove-Item -Recurse build`.

**Comment by ghost** [comment]
ok so
images are generated but
Only the second tree with "Eyes/Four" generate images with eyes
The first tree with "Eyes" generate none eyes at all

**Comment by ghost** [comment]
![tgj](https://user-images.githubusercontent.com/103067553/162593154-5b70fb4f-e230-4d67-b83d-8e9fe4926b2e.PNG)

**Comment by bolshoytoster** [comment]
@ADRIhein2 could you try changing back the
```js
    layer = layerConfigs[element.split('/').slice(0, -1).join('/')];
```
to
```js
    layer = layerConfigs[element.split('/').shift()];
```
at ~line 294

Then change
```js
         elements.concat(getElements(path, `${name}/${i}`));
```
to
```js
         elements = elements.concat(getElements(path, `${name}/${i}`));
```
at ~line 109.

**Comment by ghost** [comment]
i did it 
![èuftgy](https://user-images.githubusercontent.com/103067553/162593632-fe651439-f3b4-4168-9990-2db3cf358ae8.PNG)

**Comment by bolshoytoster** [comment]
@ADRIhein2 what if you try changing
```js
    layer = layerConfigs[element.split('/').shift()];
```
back to
```js
    layer = layerConfigs[element.split('/').slice(0, -1).join('/')];
```

**Comment by ghost** [comment]
exactly the same

**Comment by bolshoytoster** [comment]
@ADRIhein2 looks like a big issue with the way hashlips generates DNA, you're probably best just having two different folders for each one while I try to fix this.

**Comment by ghost** [comment]
maybe i should try to just ad some folder instead of sub folder bc yes its obviously the problem
but thanks anyway
alright !
